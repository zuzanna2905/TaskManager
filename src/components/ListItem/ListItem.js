import React from 'react';
const itemDesign = 'flex center f6 b--black-20 bg-white-60 ba br4 w-80';

const ListItem = (props) => {
  return (
    <li className='flex'>
      <div className={itemDesign}>
        <p className='w-40'>{props.value}</p>
        <p className='w-20'>Category: {props.category}</p>
        <p className='w-20'>Priority: {props.priority}</p>
        <button className='w-20 br4 bg-black-20' onClick={() => props.handleDelete(props.id)}>Done</button>
      </div>
    </li>
  )
}

export default ListItem;
