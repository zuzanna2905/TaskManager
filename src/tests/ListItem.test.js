import {shallow} from 'enzyme';
import React from 'react';
import ListItem from '../components/ListItem/ListItem';

it('renders ListItem without crashing', () => {
    const initialProps = {
        value: 'clean room', 
        id: 2
    }
    let  wrapper = shallow(<ListItem {...initialProps}/>);
    expect(wrapper).toMatchSnapshot();
});