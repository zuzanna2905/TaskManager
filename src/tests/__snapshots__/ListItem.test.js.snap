// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ListItem without crashing 1`] = `
<li
  className="flex"
>
  <div
    className="flex center f6 b--black-20 bg-white-60 ba br4 w-90"
  >
    <p
      className="w-80 pv1 overflow-x-auto"
    >
      clean room
    </p>
    <button
      className="w-20 br4 bg-black-20"
      onClick={[Function]}
    >
      Done
    </button>
  </div>
</li>
`;
